src/query/service/src/sessions/query_ctx.rs

```diff
@@ -126,6 +120,7 @@ impl QueryContext {
             shared,
             query_settings,
             fragment_id: Arc::new(AtomicUsize::new(0)),
+            segment_locations: Arc::new(RwLock::new(Vec::new())),
```

Comment:
I think we can change a better name to indicate these `segment_locations` are generated by insertion. And add some comments.